### PR TITLE
Define colors in ThemeProvider

### PR DESCRIPTION
**Pull request by alexanbj**
Resolves GlobalDigitalLibraryio/issues#81

### CODE REVIEW

File: components/Button.js [hunk 1]
```diff
@@ -9,9 +9,13 @@
...
 
 const Button = styled.button`
-  color: #fff;
+  color: ${props => props.theme.grays.white};
   border-radius: 4px;
-  background: linear-gradient(180deg, #5cbc80 0%, #359258 100%);
+  background: linear-gradient(
+    180deg,
+    ${props => props.theme.supports.greenPrimary} 0%,
+    #056d2c 100%
```
Comment (runarfu) [inline components/Button.js]: Both these colors are referred to, under "gradient 2" in the styleguide. Should they be called `props.theme.gradients.gradient2.start` and `props.theme.gradients.gradient2.end` or something similar instead?

File: components/Card.js [hunk 2]
```diff
@@ -81,13 +81,12 @@ const CardDropdownItem = styled.a`
   }
 
   &:not(:last-child) {
-    border-bottom: 1px solid #e8e3e3;
+    border-bottom: 1px solid ${props => props.theme.grays.platinum};
   }
 `;
 
 const CardAction = styled.a`
-  ${fontSize} color: #20588f;
```
Comment (runarfu) [inline components/Card.js]: Just to be sure: Should this color be removed?

File: components/Hero.js [hunk 3]
```diff
@@ -11,10 +11,12 @@ import Box from './Box';
 const Hero = Box.extend`
   background: ${props =>
     props.colorful
-      ? 'linear-gradient(90deg, #014C90 0%, #4EAD81 100%)'
+      ? 'linear-gradient(135deg, #004b91 0%, #5abc7f 100%)'
```
Comment (runarfu) [inline components/Hero.js]: Same comment as above - call these `props.theme.gradients.gradient4.start` and `props.theme.gradients.gradient4.end` or something similar?

File: components/Navbar/Navbar.js [hunk 4]
```diff
@@ -15,19 +15,17 @@ import Container from '../Container';
...
-const color = '#fff';
-
 const Nav = styled.nav`
   position: relative;
   ${responsiveStyle(
     'min-height',
     'minHeight',
-  )} background: linear-gradient(0deg, #4884be 0%, #20588f 100%);
+  )} background: linear-gradient(0deg, #4884be 0%, ${props =>
+  props.theme.primaries.primary} 100%);
```
Comment (runarfu) [inline components/Navbar/Navbar.js]: Same comment here about named gradients.

File: components/Card.js (hunk 2 above)
Comment (alexanbj) [reply components/Card.js]: Yes. The color of anchor tags is defined in the global styles, so this is unnecessary duplication

**Comment by alexanbj** [comment]
Currently the gradients aren't reused anywhere, so for now I think it's okay that they are defined directly in their components. If the same gradients are reused however, they will definitely be added to the themeprovider

**Pull request merged**